### PR TITLE
Add better auto-closing UX for Liquid pairs

### LINKED ISSUE #202: Better auto-closing pairs
**Is your feature request related to a problem? Please describe.**
Echo of Shopify/theme-check-vscode#108

> In PHPStorm Liquid the spaces are automatically added, this also kills my productivity :smile:

**Describe the solution you'd like**
Better auto space pairs.

### PR DESCRIPTION
**Pull request by charlespwd**
## What are you adding in this PR?

- Add better auto-closing UX for Liquid pairs
- Use typescript for the VS Code language config

https://github.com/Shopify/theme-tools/assets/4990691/0469992d-654e-46b1-b928-b15c92d3d52e

Fixes #202
Fixes Shopify/theme-check-vscode#108

## What did you learn?

- [OnTypeFormatting](https://code.visualstudio.com/api/language-extensions/programmatic-language-features#incrementally-format-code-as-the-user-types) is a thing in the LSP.
- It's off by default in VS Code, but can be changed by the extension via [`configuationDefaults`](https://code.visualstudio.com/api/references/contribution-points#contributes.configurationDefaults)
- `autocloseBefore` and `autoclosingPairs` don't go well together if one of the closing pair character is also a autocloseBefore character (space)
- This shit is complicated.

## Before you deploy

- [x] I included a minor bump `changeset`
- [x] My feature is backward compatible

### CODE REVIEW

File: packages/theme-language-server-common/src/formatting/OnTypeFormattingProvider.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        }
+      }
+
+      // here we fix {%| %} with {% | %}
+      case '%': {
+        const chars = currentLine.slice(charIdx - 1, charIdx + 4);
+        if (chars === '{% %}') {
+          return [TextEdit.insert(Position.create(line, charIdx + 1), ' ')];
+        }
+      }
```
Comment (albchu) [inline packages/theme-language-server-common/src/formatting/OnTypeFormattingProvider.ts:79]: Smallest of nits but these two are basically the same case. Could just `||` the conditional if we want to smash them together.

File: packages/vscode-extension/scripts/language-configuration.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { LanguageConfiguration } from 'vscode';
+import { indentationRules, IndentationRulesJSON } from './indentation-rules';
+import { onEnterRules, OnEnterRuleJSON } from './on-enter-actions';
+
+// interface LanguageConfigurationJSON
+//   extends Omit<LanguageConfiguration, 'onEnterRules' | 'indentationRules'> {
+//   onEnterRules?: OnEnterRuleJSON[];
+//   indentationRules?: IndentationRulesJSON;
+// }
```
Comment (albchu) [inline packages/vscode-extension/scripts/language-configuration.ts:9]: 👀

**Review by albchu — APPROVED** [review]
LGTM. Some tiny nits called out but otherwise good to go.

File: packages/vscode-extension/scripts/language-configuration.ts (hunk 2 above)
Comment (charlespwd) [reply packages/vscode-extension/scripts/language-configuration.ts:9]: Oh yeah forgot about that :P

**Pull request merged**